tools/precommit.py

```diff
@@ -117,28 +119,44 @@ def build(buildtype, args=[]):
     ex.check_run_cmd('./tools/build.py', ['--buildtype=' + buildtype] + args)
 
 
+def get_os_dependency_exclude_module(exclude_module):
+    os_dependency_module = {}
+    all_module = set(exclude_module['all'])
+    for os_name in exclude_module.keys():
+        if not os_name == 'all':
+            os_dependency_module[os_name] = \
+              list(all_module | set(exclude_module[os_name]))
+    return os_dependency_module
+
+
 option = parse_option()
 config = get_config()
-if len(config['module']['exclude']) > 0 :
-    include_module = ','.join(config['module']['exclude'])
-    include_module = ['--iotjs-include-module=' + include_module]
-else:
-    include_module = []
+os_dependency_module = \
+  get_os_dependency_exclude_module(config['module']['exclude'])
+
+for os_name in os_dependency_module:
+    if os_dependency_module[os_name]:
+        os_dependency_module[os_name] = \
+        ['--iotjs-include-module=' + ','.join(os_dependency_module[os_name])]
```

Comment:
I've never really understood this part of code, not even in its original form. Whatever we give in the config file in the `['module']['exclude']` section, will be passed on to the build script as a module to be included with `--iotjs-include-module`. What is the rationale behind that? It is a bit counter-intuitive, so at least some comments explaining it would be helpful.